ui/lib/config-ui/addon/components/messages/message-expiration-date-form.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { localDateTimeString } from 'vault/models/config-ui/message';
+
+/**
+ * @module Messages::MessageExpirationDateForm
+ * Messages::MessageExpirationDateForm components are used to display list of messages.
+ * @example
+ * ```js
+ * <Messages::MessageExpirationDateForm @message={{this.message}} @attr={{attr}} />
```

Comment:
Instead of having this component manage setting the model value, could you just pass the attribute and an "onChange" function that passes back the end time as either an empty string or a timestamp? I don't think you'd need a component file in that case 🤔 
That way this component acts more as a fancy input, and all of the model updates happen in the `create-and-edit-` form

Reply:
I did something similar at one point in this PR, but removed it since I thought it was easier to understand this way since you're able to see the logic in the template rather than navigating back to the `create-and-edit-` to see the onChange action. 🤷‍♀️ I think we'll need a component no matter what since the groupValue and formDateTime tracked properties are initially set in the constructor based on whether an endTime exists. An argument can be made that this could be in the `create-and-edit-` too, but I feel like this should be role of the `message-expiration-date-form`

Reply:
I'll make note of this as a possible improvement, but don't think it should be a blocker for merging this in the sidebranch. The next ticket I'll work on is the edit so I'll try addressing this there.

Reply:
Definitely. My suggestion was based on handling the input similar to the pattern of how other form fields work. You definitely have more context here! I find attribute changes are easier to follow if all input components are responsible for is returning the selected/inputed value. In this case, that'd be an ISO timestamp string.